src/jumpstart/JumpstartAddAssets.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useRef } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { ScrollView, StyleSheet, Text } from 'react-native'
+import { SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context'
+import AddAssetsBottomSheet, { AddAssetsAction } from 'src/components/AddAssetsBottomSheet'
+import BackButton from 'src/components/BackButton'
+import { BottomSheetRefType } from 'src/components/BottomSheet'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import CustomHeader from 'src/components/header/CustomHeader'
+import { CICOFlow, FiatExchangeFlow, fetchExchanges } from 'src/fiatExchanges/utils'
+import Palm from 'src/icons/Palm'
+import WaveCurve from 'src/icons/WaveCurve'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { isAppSwapsEnabledSelector } from 'src/navigator/selectors'
+import { userLocationDataSelector } from 'src/networkInfo/selectors'
+import { useSelector } from 'src/redux/hooks'
+import Colors from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import { TokenActionName } from 'src/tokens/types'
+
+export default function JumpstartAddAssets() {
+  const addAssetsBottomSheetRef = useRef<BottomSheetRefType>(null)
+
+  const { t } = useTranslation()
+  const insets = useSafeAreaInsets()
+
+  const userLocation = useSelector(userLocationDataSelector)
+  const isSwapEnabled = useSelector(isAppSwapsEnabledSelector)
+
+  const asyncExchanges = useAsync(async () => {
+    try {
+      return await fetchExchanges(userLocation.countryCodeAlpha2)
+    } catch (error) {
+      return []
+    }
+  }, [])
+
+  const handleShowAddFunds = () => {
+    // TODO analytics
+    addAssetsBottomSheetRef.current?.snapToIndex(0)
+  }
+
+  const addAssetsActions: AddAssetsAction[] = [
+    {
+      name: TokenActionName.Add,
+      details: t('jumpstartIntro.addFundsCelo.actionDescriptions.add'),
+      onPress: () => {
+        // TODO analytics
+        navigate(Screens.FiatExchangeCurrencyBottomSheet, { flow: FiatExchangeFlow.CashIn })
+      },
+    },
+    {
+      name: TokenActionName.Transfer,
+      details: t('jumpstartIntro.addFundsCelo.actionDescriptions.transfer'),
+      onPress: () => {
+        // TODO analytics
+        navigate(Screens.ExchangeQR, {
+          flow: CICOFlow.CashIn,
+          exchanges: asyncExchanges.result ?? [],
```

Comment:
Should the fallback be an empty array or `undefined`? Or is it the same?

Reply:
ah, i didn't think when i copied this from the earn stuff but on closer inspection, they are not the same but they should be. i updated `ExchangeQR` to make it the same (we currently allow rendering the exchanges bottom sheet if `exchanges` is defined but what's the point if the array is empty)